Show private activities too on calender when logged in

LenaWil opened the pull request:
Otherwise they are so hidden that it is useless. I am still working on the pr, but what do y’all think?

(I couldn’t quickly find where to set the user in the test)

Todo:
- [x] fix tests that fail currently.
- [ ] write test logging out and in with activities.
- [ ] check layout webpage. maybe edit to make clear it works on Apple packages.

LenaWil committed:
Show private activities too on calender when logged in

EricTRL commented on user_interaction/views.py:
```diff
@@ -95,7 +95,14 @@ def get_context_data(self, **kwargs):
+            Activity.objects.filter(published_date__lte=timezone.now())
+            .filter(
+                Q(type=ActivityType.ACTIVITY_PUBLIC)
+                | (Q(type=ActivityType.ACTIVITY_MEETING) & Q(organisers__members=self.request.member))
```
I'm not sure what will happen if `self.request.member = None`. Might be useful to check that condition first just to be safe. Something along the lines of (pseudocode):
```python
activities = Activity.objects.filter(PUBLIC-Q)
if member is not None:
    activities = activities.filter(MEETINGS-Q)
for activity in activities:
    ...
```

EricTRL reviewed (changes requested):
I agree with the changes, but tests should be added.

Each `TestCase` class has a `client` attribute. You can set a user there through `self.client.force_login(<user>)`. If you're working with raw requests, you can set the `Request.user` attribute directly as well.

LenaWil commented:
> You can set a user there through `self.client.force_login(<user>)`.

How do you log out an user, `self.client.force_login(None)`?

LenaWil committed:
Work on the private meetings calendar:

Add a flake8 config file with a max line length so that stops complaining
Add some pytest plugings so I can run the tests from VS code
Some formatting changes
Add if statement to check if member is none
Try to change the input of the private meetings of the calendar (not finished yet)
Add a button to subscribe to private meetings of the calendar (not finished yet)

todo: tests

LenaWil committed:
Update python in the CI to 3.10 to use a certain type hint

LenaWil committed:
Add python django to the ci

LenaWil committed:
Go back to python 3.8 and hopefully fix the issue

LenaWil committed:
Try & fail to get an icon

EricTRL commented on .github/workflows/run_tests.yml:
```diff
@@ -10,7 +10,7 @@ jobs:
     strategy:
       matrix:
-        python: ["3.8"]
+        python: ["3.10"]
```
I think we're still running Python 3.8 in production, so please keep the two identical. Most type hints are already available in 3.8. E.g. `List[str]` (with an import for `List` from Python's `typing` module) opposed to `list[str]`

LenaWil replied:
Yeah I already reverted this before you commented, but thanks for the feedback!

LenaWil committed:
rommel
